### PR TITLE
[fix flapping tests] Try circleci as CI

### PR DESCRIPTION
**Pull request by slonka**
Trying to solve: https://github.com/allegro/envoy-control/issues/47

### CODE REVIEW

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/HttpIdleTimeoutTest.kt [hunk 1]
```diff
@@ -27,57 +27,62 @@ internal class HttpIdleTimeoutTest : EnvoyControlTestConfiguration() {
...
 
-    private fun assertHasActiveConnection(container: EnvoyContainer) {
-        val http2Connections = container.admin().statValue("cluster.proxy1.upstream_cx_active")?.toInt()
-        assertThat(http2Connections).isGreaterThan(0)
+    private fun assertHasActiveConnection(
+        container: EnvoyContainer,
+        name: String
+    ) {
+        val activeConnections = container.admin().statValue("cluster.$name.upstream_cx_active")?.toInt()
+        assertThat(activeConnections).isEqualTo(1)
```
Comment (wookieJ) [inline envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/HttpIdleTimeoutTest.kt]: Why not `isGreaterThan(0)` ?

Comment (slonka) [reply envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/HttpIdleTimeoutTest.kt]: It should be 1, I do not remember why originally I wrote `>0`.

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/HttpIdleTimeoutTest.kt [hunk 2]
```diff
@@ -10,7 +10,7 @@ import pl.allegro.tech.servicemesh.envoycontrol.config.envoy.EnvoyContainer
 import java.util.concurrent.TimeUnit
 
 internal class HttpIdleTimeoutTest : EnvoyControlTestConfiguration() {
-
+    private val logger by logger()
```
Comment (lukidzi) [inline envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/HttpIdleTimeoutTest.kt]: Is it used?

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/EnvoyControlTestConfiguration.kt [hunk 3]
```diff
@@ -391,11 +391,11 @@ abstract class EnvoyControlTestConfiguration : BaseEnvoyTest() {
 
     @AfterEach
     fun cleanupTest() {
-        deregisterAllServices()
         envoyContainer1.admin().resetCounters()
         if (envoys == 2) {
             envoyContainer2.admin().resetCounters()
         }
+        deregisterAllServices()
```
Comment (lukidzi) [inline envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/EnvoyControlTestConfiguration.kt]: Why did you change the order of method calls?

Comment (slonka) [reply envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/config/EnvoyControlTestConfiguration.kt]: It seemed to me that it takes longer to rest counters than to deregister All services (our tests that were based on counters were flapping). I can revert it back.

File: envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/HttpIdleTimeoutTest.kt (hunk 2 above)
Comment (slonka) [reply envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/HttpIdleTimeoutTest.kt]: I'll remove it.

**Comment by slonka** [comment]
I'm force merging this so we can disable travis.
